packages/rx_bloc_cli/mason_templates/bricks/feature_onboarding/__brick__/lib/base/data_sources/local/users_local_data_source.dart

```diff
@@ -0,0 +1,28 @@
+{{> licence.dart }}
+
+import 'package:flutter_secure_storage/flutter_secure_storage.dart';
+
+/// Data source used to save user profile to Secure Storage
+class UsersLocalDataSource {
+  /// Key used to store if the profile is temporary
+  static const _isProfileTemporary = 'isProfileTemporary';
+
+  UsersLocalDataSource(this._secureStorage);
+
+  final FlutterSecureStorage _secureStorage;
+
+  /// Returns true if the profile is temporary, during onboarding
+  Future<bool> isProfileTemporary() async =>
+      await _secureStorage.read(key: _isProfileTemporary) == 'true';
+
+  /// Sets the profile as temporary, during onboarding
+  Future<void> setIsProfileTemporary(bool isProfileTemporary) =>
```

Comment:
Do we really need to store this flag, or we can get it from the API ?

Reply:
@StanevPrime I believe @GoranPrime  can tell us more about this.

Reply:
I believe we need a flag like this to facilitate "Resume onboarding" while following ACs.

If the user starts onboarding, then restarts the app & opens "Registration" page again, it will immediately call `/api/users/me` with their access token, and redirect them to continue where they left off. Sounds great, and my first thought was "here we can just check for the access token, and if they have it, call the endpoint to resume onboarding".

However another issue pops up.... if we just save the user's access token during onboarding and nothing else, then the next time the user restarts the app, they will get redirected straight to the DashboardPage, without this check:
[SplashBloc](https://github.com/Prime-Holding/rx_bloc/pull/915/files#diff-19f99cf38058cdc3a8f9f829985e82adf68f4b5a15ee615490deb898ef16a3f6R76)

I followed the ACs to resume onboarding when the user goes to Login -> Register page next time after restart, however another option is to move this `/api/users/me` API call (and relevant resume onboarding redirection logic) to the SplashBloc immediately, and skip Login page if the user is resuming onboarding. Then I believe we wouldn't need this `isProfileTemporary` flag.

Another option, as Georgi mentioned, would be to encode this TempUser flag in the access token itself, then check for that in the SplashBloc.

Reply:
Let's call the `/api/users/me` in the splash screen., since we need to get the profile anyway. Based on the response we can determine whether the profile is a `temporary` one

Reply:
Done